odxtools/statetransitionref.py

```diff
@@ -0,0 +1,219 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+from xml.etree import ElementTree
+
+from .basicstructure import BasicStructure
+from .dataobjectproperty import DataObjectProperty
+from .exceptions import odxassert, odxraise, odxrequire
+from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId, OdxLinkRef, resolve_snref
+from .odxtypes import ParameterValue, ParameterValueDict
+from .parameters.codedconstparameter import CodedConstParameter
+from .parameters.parameter import Parameter
+from .parameters.physicalconstantparameter import PhysicalConstantParameter
+from .parameters.tablekeyparameter import TableKeyParameter
+from .parameters.tablestructparameter import TableStructParameter
+from .parameters.valueparameter import ValueParameter
+from .snrefcontext import SnRefContext
+from .statemachine import StateMachine
+from .statetransition import StateTransition
+from .utils import dataclass_fields_asdict
+
+if TYPE_CHECKING:
+    from .statemachine import StateMachine
+    from .tablerow import TableRow
+
+
+def _resolve_in_param(
+    in_param_if_snref: Optional[str],
+    in_param_if_snpathref: Optional[str],
+    params: List[Parameter],
+    param_dict: ParameterValueDict,
+) -> Tuple[Optional[Parameter], Optional[ParameterValue]]:
+
+    if in_param_if_snref is not None:
+        path_chunks = [in_param_if_snref]
+    elif in_param_if_snpathref is not None:
+        path_chunks = in_param_if_snpathref.split(".")
+    else:
+        return None, None
+
+    return _resolve_in_param_helper(params, param_dict, path_chunks)
+
+
+def _resolve_in_param_helper(
```

Comment:
Isn't this info already in the encode state/ decode state ?

Reply:
not really. there already is a close analog in the [variant matching code](https://github.com/mercedes-benz/odxtools/blob/main/odxtools/matchingparameter.py#L80-L138), though. (the traversal of the pathref is basically identical, but what is done is different...)